src/WebJobs.Script.WebHost/Configuration/ScriptApplicationHostOptionsSetup.cs

```diff
@@ -61,6 +69,69 @@ public void Configure(string name, ScriptApplicationHostOptions options)
                 options.IsSelfHost = options.IsSelfHost;
                 options.IsStandbyConfiguration = true;
             }
+
+            options.AreZipDeploymentAppSettingsValid = ValidateZipDeploymentAppSettings();
+            options.ScmRunFromPackageBlobExists = BlobExists();
```

Comment:
>BlobExists(); [](http://example.com/codeflow?start=50&length=13)

Since BlobExists() needs to be called only if SCM_RUN_FROM_PACKAGE is present, Set it to false here and update later on.

Reply:
From my understanding, the assignment on L74 happens whenever we call configuration bind on options, which is on startup. So, if I set it to false here, it will remain false until we actually call storage which is only from the codepath that calls `IsZipDeployment` or `IsFileSystemReadOnly`.

However, in `RunFromPackageContext#IsRunFromPackage` we directly check against `options.ScmRunFromPackageBlobExists`. That means we implicitly rely on having a `IsZipDeployment` / `IsFileSystemReadOnly` call before specialization (otherwise it will just always be false), which I'm not sure happens today.

Reply:
Maybe I can delay the assignment using a Lazy bool?

Reply:
Sorry I meant still doing in the same method, but not doing it before we check if the app is using WEBSITE_RUN_FROM_PACKAGE or not.

something like

Instead of 
options.ScmRunFromPackageBlobExists = BlobExists();
options.IsZipDeployment = IsZipDeployment(options);

change to

options.IsZipDeployment = IsZipDeployment(options, options.ScmRunFromPackageBlobExists);

```
bool IsZipDeployment(bool scmRunFromPackageBlobExists)
{
scmRunFromPackageBlobExists = false;
bool runFromPkgConfigured = IsValidZipSetting(_environment.GetEnvironmentVariable(AzureWebsiteZipDeployment)) ||
                IsValidZipSetting(_environment.GetEnvironmentVariable(AzureWebsiteAltZipDeployment)) ||
                IsValidZipSetting(_environment.GetEnvironmentVariable(AzureWebsiteRunFromPackage));

   if (runFromPkgConfigured) {return true;}

bool scmRunFromPkgConfigured = IsValidZipSetting(_environment.GetEnvironmentVariable(ScmRunFromPackage)) && BlobExists(_environment.GetEnvironmentVariable(ScmRunFromPackage);

scmRunFromPackageBlobExists = scmRunFromPkgConfigured;

return scmRunFromPkgConfigured;
}
```

Basically BloblExists()  needs to be called only if runFromPkgConfigured is false. Doesn't necessarily have to use the same pattern :)

Reply:
Thanks for clarifying, I understand now what you meant. Agreed, this is better!